### PR TITLE
Update julia from 1.7.3 to 1.8.0

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-cask-pr`.

### CODE REVIEW

File: Casks/julia.rb [hunk 1]
```diff
@@ -1,8 +1,12 @@
 cask "julia" do
-  version "1.7.3"
-  sha256 "67799ec06cf57bca80a8a1c6e5a170485d3bfc0461b217699b0730da37d0656f"
+  arch arm: "aarch64", intel: "x64"
+  midfix = on_arch_conditional arm: arch
 
-  url "https://julialang-s3.julialang.org/bin/mac/x64/#{version.major_minor}/julia-#{version}-mac64.dmg"
+  version "1.8.0"
+  sha256 arm:   "8e113a42fe4eac7b75023136674452f6f09590717755417cd137258707e96540",
+         intel: "29880bb57614c9c597edf5384a351b3f0299fa097b9eb62cd640c93cde58677b"
```
Comment (carlocab) [inline Casks/julia.rb:7]: @Rylan12 do we support this syntax yet? I've forgotten where we are on this.

Comment (p-linnane) [reply Casks/julia.rb:7]: @carlocab I've been keeping an eye on this. https://github.com/Homebrew/brew/pull/13702 has been merged but still waiting for the rubocop PR to be merged (https://github.com/Homebrew/brew/pull/13703).

Comment (Rylan12) [reply Casks/julia.rb:7]: Not yet, we need a new brew tag first. My plan is to make one later today

**Review by miccal — APPROVED** [review]

**Comment by jan-xyz** [comment]
This produces a broken link for arm64: `https://julialang-s3.julialang.org/bin/mac/aarch64/1.8/julia-1.8.0-macaarch6464.dmg` (see double `64`). I tried to understand how the syntax works but don't know where the double 64 comes from. I suspect we can remove it from the link template, but I don't know if that breaks it for Intel?

**Comment by miccal** [comment]
Thanks @jan-xyz -- this will be fixed by https://github.com/Homebrew/homebrew-cask/pull/130042

**Comment by adavidzh** [comment]
I also noticed the `6464` issue and made https://github.com/Homebrew/homebrew-cask/pull/130044.

**Comment by jan-xyz** [comment]
I think that fix would've broken the intel installation 🤔

**Pull request merged**